File: app/src/main/java/com/example/amazonprimeclone/fragments/home/more/see_all_series/SeeAllSeriesViewModel.java
package com.example.amazonprimeclone.fragments.home.more.see_all_series;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.SavedStateHandle;
import androidx.lifecycle.ViewModel;

import com.example.amazonprimeclone.data.remote.Constants;
import com.example.amazonprimeclone.modal.SearchSeries;
import com.example.amazonprimeclone.modal.SeriesResponseResults;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SeeAllSeriesViewModel extends ViewModel {

    private final MutableLiveData<List<SeriesResponseResults>> allSeriesList;
    private final String id;
    public final String title;

    public SeeAllSeriesViewModel(SavedStateHandle savedStateHandle){
        allSeriesList = new MutableLiveData<>();
        id = savedStateHandle.get("id");
        title = savedStateHandle.get("title");

        getAllSeries();
    }

    public MutableLiveData<List<SeriesResponseResults>> getAllSeriesList() {
        return allSeriesList;
    }

    private void getAllSeries() {

        Call<SearchSeries> movieResponseCall = Constants.retrofitService.getCategoriesResultsSeries(id);
        movieResponseCall.enqueue(new Callback<SearchSeries>() {
            @Override
            public void onResponse(@NonNull Call<SearchSeries> call, @NonNull Response<SearchSeries> response) {
                SearchSeries searchSeries = response.body();
                //add to list
                if(searchSeries!=null){
                    allSeriesList.setValue(searchSeries.getResults());
                }

            }

            @Override
            public void onFailure(@NonNull Call<SearchSeries> call,@NonNull Throwable t) {

            }
        });

    }
}

File: app/src/main/java/com/example/amazonprimeclone/fragments/home/more/see_all_series/SeeAllSeriesViewModel.kt
package com.example.amazonprimeclone.fragments.home.more.see_all_series

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.SavedStateHandle
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.amazonprimeclone.modal.SeriesResponseResults
import com.example.amazonprimeclone.retrofit.repository.RetrofitMovieRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class SeeAllSeriesViewModel @Inject constructor(
    savedStateHandle: SavedStateHandle,
    private val retrofitMovieRepository: RetrofitMovieRepository
): ViewModel(){

    private val allSeriesList: MutableLiveData<List<SeriesResponseResults>> = MutableLiveData()
    private var id: String? = null
    var title: String? = null

    init {
        id = savedStateHandle.get<String>("id")
        title = savedStateHandle.get<String>("title")
        getAllSeries()
    }

    fun getAllSeriesList(): MutableLiveData<List<SeriesResponseResults>> {
        return allSeriesList
    }

    private fun getAllSeries() = viewModelScope.launch{
        val response = retrofitMovieRepository.getCategoriesResultsSeries(id)
        if(response!=null){
            allSeriesList.postValue(response.results)
        }
    }

}